domain/src/main/java/com/teamagam/gimelgimel/domain/messages/CreateAndQueueMessageInteractor.java

```diff
@@ -5,38 +5,29 @@
 import com.teamagam.gimelgimel.domain.base.interactors.DataSubscriptionRequest;
 import com.teamagam.gimelgimel.domain.config.Constants;
 import com.teamagam.gimelgimel.domain.messages.entity.ChatMessage;
-import com.teamagam.gimelgimel.domain.messages.repository.MessagesRepository;
-import com.teamagam.gimelgimel.domain.notifications.repository.MessageNotifications;
+import com.teamagam.gimelgimel.domain.messages.entity.OutGoingMessagesQueue;
 import com.teamagam.gimelgimel.domain.user.repository.UserPreferencesRepository;
 import io.reactivex.Observable;
 import java.util.Collections;
 
-public abstract class SendMessageInteractor extends BaseDataInteractor {
+public abstract class CreateAndQueueMessageInteractor extends BaseDataInteractor {
```

Comment:
It should be named CreateAnd*Enqueue*....
It might even be fine if you emit the create, since it's obvious it'll create it if needed.
Maybe even - EnqueueMessageForSendingInteractor